ui/apps/platform/src/services/ComplianceResultsService.ts

```diff
@@ -46,11 +46,19 @@ export type ListComplianceCheckClusterResponse = {
  */
 export function getComplianceProfileCheckResult(
     profileName: string,
-    checkName: string
+    checkName: string,
+    page: number,
+    perPage: number
```

Comment:
Here is a question for the mental slow cooker, not necessarily for this contribution.

As the number of arguments of a service function increases, what are pro and con:
* Two string arguments like `profileName` and `checkName` which correspond to order in endpoint address but two number arguments `page` and `perPage` for which order is more arbitrary (although consistent between many functions).
    Compare to following `getComplianceProfileResults(profileName, sortOption, page, perPage)` function.
* One auxiliary object argument whose property names correspond to PatternFly semantics.
    * `getComplianceProfileCheckResult(profileName, checkName, { page, perPage })`
    * `getComplianceProfileResults(profileName, { page, perPage, sortOption })`

```ts
export type Whatchamacallit = {
    page: number;
    perPage: number;
    sortOption?: ApiSortOption;
};
```

With a helper function that converts `offset: (page - 1) * perPage, limit: perPage` for request params:

https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/services/types.ts#L3-L7

```ts
export type Pagination = {
    offset: number;
    limit: number;
    sortOption?: ApiSortOption;
};
```

Reply:
Interesting thought. It has felt weird to me having the "critical" (profileName, checkName, etc) alongside the "non-critical" (pagination, searchField, etc) in the list of parameters. Especially when the "non-critical" that I just described seem to be included everywhere.

I'll dabble with the idea as I implement sorting on this table